src/components/withKeyboardState.js

```diff
@@ -15,32 +14,24 @@ const keyboardStateProviderPropTypes = {
     children: PropTypes.node.isRequired,
 };
 
-class KeyboardStateProvider extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            isKeyboardShown: false,
-        };
-    }
-
-    componentDidMount() {
-        this.keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', () => {
-            this.setState({isKeyboardShown: true});
+function KeyboardStateProvider(props) {
+    const {children} = props;
```

Comment:
nit: why not destructure directly in the function signature? not a blocker though, just personal preference :D